src/aglet/target.nim

```diff
@@ -34,20 +42,61 @@ proc height*(target: Target): int {.inline.} =
   ## Returns the height of the target.
   target.size.y
 
-proc clearColor*(target: Target, color: Rgba32f) {.inline.} =
-  ## Clear the target's color with a solid color.
-  target.use()
-  target.gl.clearColor(color.r, color.g, color.b, color.a)
-
-proc clearDepth*(target: Target, depth: float32) {.inline.} =
-  ## Clear the target's depth buffer with a single value.
-  target.use()
-  target.gl.clearDepth(depth)
-
-proc clearStencil*(target: Target, stencil: int32) {.inline.} =
-  ## Clear the target's stencil buffer with a single value.
+proc scissor(target: Target, scissor: Option[Recti]) =
+  if scissor.isSome():
+    let scissor = scissor.get
+    target.gl.scissor(scissor.x, scissor.y, scissor.width, scissor.height)
+  else:
+    target.gl.scissor(0, 0, target.size.x, target.size.y)
+
+proc defaultClearParams*(): ClearParams =
+  ClearParams(
+    color: rgba(0.0, 0.0, 0.0, 1.0).some(),
+    depth: some(1.0.float32),
+    stencil: int32.none(),
+    scissor: Recti.none()
+  )
+
+proc withColor*(clearParams: ClearParams, color: Rgba32f): ClearParams =
```

Comment:
As an optimization (but unfortunately not a correctness measure), the first parameter can be `sink`, like:
```nim
proc withColor*(clearParams: sink ClearParams, color: Rgba32f): ClearParams =
```
Unfortunately this doesn't really give us any extra correctness guarantees, but at least the compiler will have a hint that `clearParams` should be moved into the procedure.

Reply:
Ah, neat! I was wondering what the Nim version of this was.